ext/node/resolution.rs

```diff
@@ -27,6 +27,34 @@ pub enum NodeModuleKind {
   Cjs,
 }
 
+/// Checks if the resolved file has a corresponding declaration file.
+pub fn path_to_declaration_path(
+  path: PathBuf,
+  referrer_kind: NodeModuleKind,
+) -> PathBuf {
+  let lowercase_path = path.to_string_lossy().to_lowercase();
+  if lowercase_path.ends_with(".d.ts")
+    || lowercase_path.ends_with(".d.cts")
+    || lowercase_path.ends_with(".d.ts")
+  {
+    return path;
+  }
+  let specific_dts_path = match referrer_kind {
+    NodeModuleKind::Cjs => path.with_extension("d.cts"),
```

Comment:
Do we need to watch out for paths with a dot in it like `something.service` here?

Reply:
We definitely should :D

Reply:
Actually, I'm not sure - since these package are types only they are expected to have `.d.ts` (or similar)?

Reply:
I think if the subpath was something like `something.service` then this would change that to `something.d.cts` instead of `something.service.d.cts`?

Reply:
Yeah, good point, I'll add some unit tests to this helper function